presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java

```diff
@@ -98,6 +100,7 @@ public JoinNode(
         this.criteria = ImmutableList.copyOf(criteria);
         this.leftOutputSymbols = ImmutableList.copyOf(leftOutputSymbols);
         this.rightOutputSymbols = ImmutableList.copyOf(rightOutputSymbols);
+        this.canSkipOutputDuplicates = canSkipOutputDuplicates;
```

Comment:
`canSkipOutputDuplicates = true` only makes sense if the only columns being output come from the left side of the join (or are equi-join keys in the case of an inner join). Also, this doesn't work for RIGHT or FULL join, so we should validate we're not creating an inconsistent node.

Also, given the semantics of this property, I'm not sure `canSkipOutputDuplicates` is the best name. It'd be more accurate to call it "outputsFirstMatchOnly" or something similar.

Reply:
Like I suggested in https://github.com/prestosql/presto/issues/6005, we need to sort it out before we start optimizing other plans based on `canSkipOutputDuplicates` property. 
In the case captured by the `OptimizeOutputDuplicatesInsensitiveJoinRule `, the aggregation remains in the plan, so the result is the same, no matter if the join is or is not eventually optimized. However, before we add other transformations depending on join dropping duplicates, we need to find a way to ensure that the execution will follow - that is, that the join will not flip.

Reply:
> However, before we add other transformations depending on join dropping duplicates, we need to find a way to ensure that the execution will follow - that is, that the join will not flip.

This looks like committing to some concrete implementation of Join with it's properties (e.g cannot be flipped). IIRC some planners actually plan first with abstract node types and only then have rules that choose concrete implementation.